### PR TITLE
CCv0: Merge main into CCv0 branch 

### PR DESCRIPTION
**Pull request by stevenhorsman**
Merge remote-tracking branch 'upstream/main' into CCv0

Fixes: https://github.com/kata-containers/tests/issues/5574
Depends-on: github.com/https://github.com/kata-containers/kata-containers/pull/6559

Signed-off-by: stevenhorsman <steven@uk.ibm.com>

### CODE REVIEW

**Comment by fidencio** [comment]
Cloud Hypervisor TDX CI passed but didn't report back: http://10.112.240.208:8080/job/tests-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/406/

File: .ci/install_qemu.sh [hunk 1]
```diff
@@ -233,27 +87,36 @@ main() {
...
 			;;
 	esac
 
-	# Strip "https://" to clone into $GOPATH
-	gopath_qemu_repo=${QEMU_REPO_URL/https:\/\//}
-	# These variables are used by static-build scripts
-	export qemu_version="${CURRENT_QEMU_VERSION}"
-	export qemu_repo="${QEMU_REPO_URL}"
-	run
+	case ${QEMU_ARCH} in
```
Comment (fidencio) [inline .ci/install_qemu.sh:98]: And here some logic to call that function mentioned above should be added.

**Comment by stevenhorsman** [comment]
On ubuntu 20.04 (with QEMU IIRC) we're seeing consistent failures with one of the tests:
```
not ok 1 Run a job to completion
12:00:52 # (from function `waitForProcess' in file /tmp/jenkins/workspace/kata-containers-CCv0-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/lib/common.bash, line 107,
12:00:52 #  in test file k8s-job.bats, line 29)
12:00:52 #   `waitForProcess "$wait_time" "$sleep_time" "$cmd"' failed
12:00:52 # INFO: k8s configured to use runtimeclass
12:00:52 # job.batch/job-pi-test created
12:00:52 #   Normal  SuccessfulCreate  0s    job-controller  Created pod: job-pi-test-9cx4l
12:00:52 # pod "job-pi-test-9cx4l" deleted
12:00:52 # NAME                READY   STATUS    RESTARTS   AGE
12:00:52 # job-pi-test-8mhk4   0/1     Pending   0          0s
12:00:52 # job.batch "job-pi-test" deleted
```
It is working on the firecracker and fedora. Does anyone have any ideas if there were any changes to this that might have gone wrong in the merge?

**Comment by arronwy** [comment]
TDX CI for cloud hypervisor with `offline_fs_kbc`
```
22:09:34 INFO: Create CC image configured with AA_KBC=offline_fs_kbc
22:09:34 build image
22:09:34 /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/cc-rootfs-image/builddir
22:09:34 INFO: Build image
22:09:34 INFO: image os: ubuntu
22:09:34 INFO: image os version: latest
22:09:34 Creating rootfs for ubuntu
22:09:34 /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/rootfs.sh -o 3.2.0-alpha0-c7ef425f46391152017f8304c73a651cd71278cf -r /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/cc-rootfs-image/builddir/rootfs-image/ubuntu_rootfs ubuntu
22:09:34 INFO: rootfs_lib.sh file found. Loading content
22:09:34 INFO: Detecting agent rust version
22:09:34 INFO: Get rust version from /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/../../../versions.yaml
22:09:34 Required rust version: 1.66.0
22:09:34 /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
22:09:34 /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
22:09:43 make[1]: *** [tools/packaging/kata-deploy/local-build/Makefile:60: cc-rootfs-image-tarball-build] Error 137
22:09:43 make[1]: Leaving directory '/home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers'
22:09:43 make: *** [tools/packaging/kata-deploy/local-build/Makefile:147: cc-rootfs-image-tarball] Error 2
22:09:43 [install_kata_image.sh:188] ERROR: sudo -E PATH=/var/lib/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/var/lib/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/bin:/usr/local/sbin make cc-rootfs-image-tarball
22:09:43 [install_kata.sh:42] ERROR: .ci/install_kata_image.sh 
22:09:43 [jenkins_job_build.sh:252] ERROR: ci/setup.sh
22:09:43 Failed at /home/jenkins/workspace/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/tests/.ci/ci_entry_point.sh +93: .ci/jenkins_job_build.sh "${repo_to_test}" -t "${TEST_TYPE}"
22:09:53 Process leaked file descriptors. See https://www.jenkins.io/redirect/troubleshooting/process-leaked-file-descriptors for more information
22:09:53 Build step 'Execute shell' marked build as failure
```

**Comment by arronwy** [comment]
TDX CI for Qemu with `cc_kbc_tdx`:
```
09:19:45 INFO: Create CC image configured with AA_KBC=cc_kbc_tdx
09:19:45 build image
09:19:45 /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/cc-tdx-rootfs-image/builddir
09:19:45 INFO: Build image
09:19:45 INFO: image os: ubuntu
09:19:45 INFO: image os version: latest
09:19:45 Creating rootfs for ubuntu
09:19:45 /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/rootfs.sh -o 3.2.0-alpha0-5b0511264e1044f43f2cf9e7526be2da4ba20d89 -r /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/cc-tdx-rootfs-image/builddir/rootfs-image/ubuntu_rootfs ubuntu
09:19:45 INFO: rootfs_lib.sh file found. Loading content
09:19:45 INFO: Detecting agent rust version
09:19:45 INFO: Get rust version from /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/../../../versions.yaml
09:19:45 Required rust version: 1.66.0
09:19:45 /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
09:19:45 /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
...
...
09:22:25    Compiling openidconnect v2.5.1
09:22:27    Compiling sigstore v0.3.3
09:22:31    Compiling kata-agent v0.1.0 (/kata-containers/src/agent)
09:22:47 make[1]: *** [tools/packaging/kata-deploy/local-build/Makefile:60: cc-tdx-rootfs-image-tarball-build] Error 137
09:22:47 make[1]: Leaving directory '/home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/kata-containers'
09:22:47 make: *** [tools/packaging/kata-deploy/local-build/Makefile:159: cc-tdx-rootfs-image-tarball] Error 2
09:22:47 [install_kata_image.sh:188] ERROR: sudo -E PATH=/var/lib/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/var/lib/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/bin:/usr/local/sbin make cc-tdx-rootfs-image-tarball
09:22:47 [install_kata.sh:42] ERROR: .ci/install_kata_image.sh 
09:22:47 [jenkins_job_build.sh:252] ERROR: ci/setup.sh
09:22:47 Failed at /home/jenkins/workspace/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/tests/.ci/ci_entry_point.sh +93: .ci/jenkins_job_build.sh "${repo_to_test}" -t "${TEST_TYPE}"
09:22:57 Process leaked file descriptors. See https://www.jenkins.io/redirect/troubleshooting/process-leaked-file-descriptors for more information
09:22:57 Build step 'Execute shell' marked build as failure
```

**Comment by stevenhorsman** [comment]
@ryansavino @UnmeshDeodhar - I'm not sure if you enabled the job or something, but it looks like the failing SNP job is back and the sev one has disappeared like we saw last month?

**Comment by UnmeshDeodhar** [comment]
@stevenhorsman i have disabled the SNP job again. Sorry I am trying to work on SNP test that is why it was enabled.

**Comment by fidencio** [comment]
I've retriggered the TDX tests.  Please, wait till we have a green run before merging it.

**Comment by fidencio** [comment]
@stevenhorsman, we're not even getting to the tests part in the TDX CI as the CI is struggling to build the components due to the network issues.

I'm giving the green light to go ahead, merge it, have the components cached, and I will deal with any possible breakage on my side.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

File: .ci/configure_containerd_for_kata.sh [hunk 2]
```diff
@@ -9,6 +9,10 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+if [ "$KATA_BUILD_CC" == "yes" ]; then
+	PREFIX="${PREFIX:-/opt/confidential-containers}"
+fi
+PREFIX="${PREFIX:-/opt/kata}"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply .ci/configure_containerd_for_kata.sh:15]: Oops, there's no need to address this one.

File: .ci/configure_crio_for_kata.sh [hunk 3]
```diff
@@ -11,6 +11,10 @@ set -o pipefail
 
 source /etc/os-release || source /usr/lib/os-release
 
+if [ "$KATA_BUILD_CC" == "yes" ]; then
+	PREFIX="${PREFIX:-/opt/confidential-containers}"
+fi
+PREFIX="${PREFIX:-/opt/kata}"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply .ci/configure_crio_for_kata.sh:17]: Neither this one.

**Pull request merged**